packages/python-packages/apiview-copilot/src/_apiview_reviewer.py

```diff
@@ -90,3 +119,179 @@ def _retrieve_static_guidelines(self, language, include_general_guidelines: bool
                 items = json.loads(f.read())
                 language_guidelines.extend(items)
         return general_guidelines + language_guidelines
+    
+    def _get_filter_expression(self) -> str:
+        """
+        Returns the filter expression for the given language.
+        """
+        if self.language == "all":
+            return "lang eq '' or lang eq null"
+        else:
+            return f"lang eq '{self.language}' or lang eq '' or lang eq null"
+
+    def _search_guidelines(self, query: str) -> List[object]:
+        credential = DefaultAzureCredential()
+        search_name = os.getenv("AZURE_SEARCH_NAME")
+        search_endpoint = f"https://{search_name}.search.windows.net"
+        client = SearchClient(endpoint=search_endpoint, index_name="guidelines-index", credential=credential)
+        result = list(client.search(
+            search_text=query,
+            top=10,
+            filter=self._get_filter_expression(),
+            semantic_configuration_name="archagent-semantic-search-guidelines",
+            query_type=QueryType.SEMANTIC,
+            query_caption=QueryCaptionType.EXTRACTIVE,
+            vector_queries=[
+                VectorizableTextQuery(
+                    text=query,
+                    fields="text_vector"                    
+                )
+            ]))
+        return result
+
+    def _search_examples(self, chunk: Section) -> List[object]:
+        credential = DefaultAzureCredential()
+        search_name = os.getenv("AZURE_SEARCH_NAME")
+        search_endpoint = f"https://{search_name}.search.windows.net"
+        client = SearchClient(endpoint=search_endpoint, index_name="examples-index", credential=credential)
+        query = str(chunk)
+        return list(client.search(
+            search_text=query,
+            top=10,
+            filter=self._get_filter_expression(),
+            semantic_configuration_name="archagent-semantic-search-examples",
+            query_type=QueryType.SEMANTIC,
+            query_caption=QueryCaptionType.EXTRACTIVE,
+            vector_queries=[
+                VectorizableTextQuery(
+                    text=query,
+                    fields="text_vector"
+                )
+            ]))
+
+    def _retrieve_guidelines_from_search(self, chunk: Section) -> List[object]:
```

Comment:
We should probably eventually write some (not evals) tests for our reviewer code. Not blocking, though

Reply:
Agree.